src/hooks/useCart/useCart.tsx

```diff
@@ -323,8 +324,15 @@ export const useCart = (
         setCheckoutResult(transactionResponse);
         setCartState("PURCHASED");
       } catch (e) {
+        // backend will throw general duplicate identifier message
+        if (
+          e.message === "Invalid Purchase Request: Duplicate identifier inputs"
+        ) {
+          setError(new Error(ERROR_MESSAGE.DUPLICATE_IDENTIFIER_INPUT));
+        } else {
+          setError(new Error(ERROR_MESSAGE.SERVER_ERROR));
+        }
         setCartState("DEFAULT");
-        setError(new Error("Couldn't checkout, please try again later"));
```

Comment:
I think we need the error to be thrown after setting the cart state:
https://github.com/rationally-app/mobile-application/pull/129/files

Reply:
this is actually just setting the error, not throwing it. would this be fine?

Reply:
Oh like it needs to be set after the cart state haha

Reply:
okay got it and changed!